tests/src/test/scala/com/nvidia/spark/rapids/GpuPartitioningSuite.scala

```diff
@@ -53,6 +53,24 @@ class GpuPartitioningSuite extends FunSuite with Arm {
       val table = packedColumn.getContiguousTable.getTable
       // The contiguous table is still responsible for closing these columns.
       (0 until table.getNumberOfColumns).map(table.getColumn).toArray
+    } else if (GpuCompressedColumnVector.isBatchCompressed(batch)) {
+      val compressedColumn = batch.column(0).asInstanceOf[GpuCompressedColumnVector]
+      val descr = compressedColumn.getTableMeta.bufferMeta.codecBufferDescrs(0)
+      val codec = TableCompressionCodec.getCodec(descr.codec)
+      withResource(codec.createBatchDecompressor(100 * 1024 * 1024L,
+        Cuda.DEFAULT_STREAM)) { decompressor =>
+        compressedColumn.getTableBuffer.incRefCount()
+        decompressor.addBufferToDecompress(compressedColumn.getTableBuffer,
+          compressedColumn.getTableMeta.bufferMeta)
+        withResource(decompressor.finishAsync()) { outputBuffers =>
+          val outputBuffer = outputBuffers.head
+          // There should be only one
+          withResource(
+            MetaUtils.getTableFromMeta(outputBuffer, compressedColumn.getTableMeta)) { table =>
+            (0 until table.getNumberOfColumns).map(i => table.getColumn(i).incRefCount()).toArray
```

Comment:
I think this is going to leak.  `extractBases` made the simplifying assumption that any columns extracted are just aliased onto some other column batch memory and therefore don't need to be closed.  However for compressed batches that is an invalid assumption.

The contract for `extractBases` should change so that callers need to always close the resulting array to cover the compression case.

Reply:
Thanks for catching that. I renamed extractBases so it would not be confused with extractBases in GpuColumnVector where the contract is different.